### PR TITLE
Derive some more ToCss implementations

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20196)

### CODE REVIEW

File: components/style/counter_style/mod.rs [hunk 1]
```diff
@@ -444,7 +444,16 @@ impl Parse for Negative {
...
 /// Empty Vec represents 'auto'
 #[derive(Clone, Debug)]
-pub struct Ranges(pub Vec<Range<Option<i32>>>);
+pub struct Ranges(pub Vec<Range<CounterBound>>);
+
+/// A bound found in `Ranges`.
+#[derive(Clone, Copy, Debug)]
+pub enum CounterBound {
+    /// An integer bound.
+    Integer(i32),
```
Comment (emilio) [inline components/style/counter_style/mod.rs]: This should be an integer and support `calc`, mind filing an issue?

File: components/style/properties/helpers.mako.rs [hunk 2]
```diff
@@ -122,6 +124,14 @@
             % if need_animatable or animation_value_type == "ComputedValue":
             #[derive(Animate, ComputeSquaredDistance)]
             % endif
+            % if not allow_empty:
```
Comment (emilio) [inline components/style/properties/helpers.mako.rs]: I feel ambivalent about this one... As long as we need to keep the `allow_empty` one this doesn't seem particularly simpler in any way.

**Review by emilio — APPROVED** [review]
r=me, with the issue filed, thanks :)

File: components/style/counter_style/mod.rs (hunk 1 above)
Comment (nox) [reply components/style/counter_style/mod.rs]: I filed https://github.com/servo/servo/issues/20197.

**Pull request merged**